### PR TITLE
Re-styling the website

### PR DESCRIPTION
**Pull request by LightworksTeam**
Summary of updates:

* Adopted for Patternfly UI toolkit (bootstrap 3)
* Unified effect cards and sizes
* Tree view for contents
* Added placeholder for tools page

Screenshot below:

screenshot-2018-4-25 lightworks user repository

### CODE REVIEW

**Comment by marcinn** [comment]
Wow, that looks awesome!

**Comment by jwrl1** [comment]
That's looking so much better than the two, rather tired, threads on the forums.

**Comment by FxSchrauber** [comment]
Looks good on Microsoft Edge.
Many Thanks.

What I noticed in my first test:

If I reduce the size of the browser window, it can happen that either the tree view becomes unreadable or no effects are displayed at all.
This might be relevant to small screens, or if the webpage was minimized within Lightworks?

> Unified effect cards and sizes
This does not seem to work with all browsers.
For example, the classic Microsoft Internet Explorer, or Opera Mini (a few months old version) on Andoid.

The website is very slow.
Is more server performance required for this?

**Comment by marcinn** [comment]
@FxSchrauber  We're talking about these issues. Slowness is significant and it will be improved soon, for sure.

**Comment by geomcd1949** [comment]
Is there a time scheduled for when the website will be moved from fx-planet to a LW-related URL, and has that URL been decided yet?
Thanks.
~George

**Comment by jwrl1** [comment]
Yes, I was wondering about that too.

**Comment by WooDzu** [comment]
Most of the pieces are in place now and @marcinn managed to optimize the performance.

* Still need to decide on the domain name.

* The URLs still need updating eg. /latest/ to /effects and removing /pages/tools to /tools etc..
It should make it easier to maintain and eventually swap the backend of the website.

I'll look into both of these things later today.

* The licence needs updating to New BSD as agreed at https://www.lwks.com/index.php?option=com_kunena&func=view&catid=7&id=175280&Itemid=81#ftop
@marcinn is this something you can do?

**Comment by marcinn** [comment]
@WooDzu I can change these urls. 

My proposals for domain:
* contrib.lwks.com
* addons.lwks.com
* extras.lwks.com
* lur.lwks.com

**Comment by marcinn** [comment]
I've changed urls.

**Comment by marcinn** [comment]
>The landing page for effects currently loads all the effects

I'd prefer pagination, because `/effects/` are ordered from the newest to the oldest and loading few instead of hundreds is fast. I like to check "what's changed", that's why I created "latest" list instead of "all" (btw: it would be nice to display time of update). With dedicated urls for pages I can share/store link to the specific page - this is not a requirement, but a nice (maybe base) functionality of WWW.  And it is quite natural for people to remember something like "an interesting effect is on the page 3" instead "it is somewhere between a 1/3 and 2/3 of the scrollbar", I think.

We can do redirection to the first category, but the view shouldn't be named "List all effects" anymore, and "the first category" view will not have such functionality as "latest effects" which I'd like to preserve.

We can introduce alternative lists like A-Z.

Summing up - the most important for me is to have possibility to browse latest/new effects list in a comfortable way, where comfortable means fast loading (also on the client side, i.e. web browser). This can be achieved by pagination or it's modification (like "infinite scroll"). Everything else is doable as you wish. 

> Can category URLs be separated from effect URLs?

All can be done, first two are easiest. Merged approach is possible but requires changes in urls generation.

**Comment by FxSchrauber** [comment]
> Shall we leave it as is or default to first available category. In other words should clicking the 'Effects' link open https://fx-planet.ovh/effects/17/ 

In this case, I would fear that some users would not look at the left tree sufficiently, and might overlook the fact that only one category is displayed.
It would be a pity if the many effects in other categories were not found.

**Comment by jwrl1** [comment]
One other thing: I took Peter's comment about the aspect ratios of the screen grabs to heart, and spent some time over the last few weeks  designing a catch-all format based on what had been posted in the test page.  You can see examples of what I have come up with at https://www.lwks.com/index.php?option=com_kunena&func=view&catid=7&id=175280&limit=15&limitstart=15&Itemid=81#177042 and comments would be much appreciated.

**Comment by FxSchrauber** [comment]
I think the main problem with different aspect ratios is that they change the row height and row width on the homepage.

Suggestion, if possible:
- Defining the most frequently used aspect ratio as standard.

- Specification of a standard height and standard width of thumbnails within the home page software (or indirectly by fixed column width and line height).

- Automatic scaling of the thumbnails according to the column width and line height, without distortions (original aspect ratio, or filling empty spaces with the background color of the website / table).

Of course, many preview images with different aspect ratios would then appear smaller than other thumbnails. The user can view the details anyway after clicking on the effect.
A later exchange of images should be possible.

**Comment by jwrl1** [comment]
Did you check the link?  A lot of the existing grabs are 16x9, but I believe that there are enough that can't be for a background into which the grab can be placed to be designed.  That way you get a stylish consistency of presentation regardless of the capture aspect ratio.

**Comment by WooDzu** [comment]
> I'd prefer pagination, because /effects/ are ordered from the newest to the oldest and loading few instead of hundreds is fast. I like to check "what's changed", that's why I created "latest" list instead of "all" (btw: it would be nice to display time of update). With dedicated urls for pages I can share/store link to the specific page - this is not a requirement, but a nice (maybe base) functionality of WWW. And it is quite natural for people to remember something like "an interesting effect is on the page 3" instead "it is somewhere between a 1/3 and 2/3 of the scrollbar", I think.

Fair enough - I'll restore pagination a send another PR for review.
To clarify, in my considerations I assumed that most of the time users would be looking for specific effects, which may be wrong assumption.
Another concern was that when someone links to "an interesting effect on page 3" after a week or two that effect would move to page 4 or 5 making the link irrelevant, which is why I wanted to avoid pagination. Not a big concern as I would believe most of the time folks would link to specific effect or category.

> All can be done, first two are easiest. Merged approach is possible but requires changes in urls generation.

I have no strong preference on this one. Let's go for `separate category URLs eg: https://fx-planet.ovh/effects/category/32` if you agree.

**Comment by FxSchrauber** [comment]
> Let's go for separate category URLs eg: https://fx-planet.ovh/effects/category/32 
Can it be ensured that after deleting an obsolete category or subcategory, this number will not be recycled for new or existing categories?
To avoid this with links to categories:
> links to "an interesting effect on page 3" after a week or two that effect would move to page 4 or 5 making the link irrelevant

Current example: [https://github.com/fx-planet/fxplanet/issues/17](https://github.com/fx-planet/fxplanet/issues/17)
